formats/json/commonMain/src/kotlinx/serialization/json/internal/lexer/AbstractJsonLexer.kt

```diff
@@ -307,6 +306,54 @@ internal abstract class AbstractJsonLexer {
      */
     abstract fun consumeKeyString(): String
 
+    private fun insideString(isLenient:Boolean, char:Char):Boolean = if (isLenient) { charToTokenClass(char) == TC_OTHER } else { char != STRING }
+
+    open fun consumeStringChunked(isLenient: Boolean, consumeChunk: (stringChunk: String) -> Unit) { // open to allow simpler implementations (i.e. StringJsonLexer)
+        val nextToken = peekNextToken()
+        if (isLenient && nextToken != TC_OTHER) return // noting to consume
+
+        if (!isLenient) {
+            consumeNextToken(STRING)
+        }
+        var currentPosition = this.currentPosition
+        var lastPosition = currentPosition
+        var char = source[currentPosition] // Avoid two range checks visible in the profiler
+        var usedAppend = false
+        while (insideString(isLenient, char)) {
+            if (!isLenient && char == STRING_ESC) { // handle escaping only in lenient mode
```

Comment:
```suggestion
            if (!isLenient && char == STRING_ESC) { // handle escaping only in non-lenient mode
```

IIRC in lenient mode you won't get here anyway as `charToTokenClass(char) == TC_OTHER` is false for STRING_ESC